### PR TITLE
Fix request failure stopping waiting for transaction

### PR DESCRIPTION
**Pull request by igor-aptos**
### Description

(Transient) failure in issuing a rest API call abruptly fails waiting for transaction. Instead, retry again, until expected time is reached.

### CODE REVIEW

File: crates/aptos-rest-client/src/lib.rs [hunk 1]
```diff
@@ -620,6 +620,9 @@ impl Client {
                         )
                     );
                 },
+                Err(err) => {
+                    debug!("Fetching error, will retry: {}", err);
```
Comment (JoshLind) [inline crates/aptos-rest-client/src/lib.rs:624]: nit: can we maybe also print the default delay to output how long we'll wait until we retry?

**Review by JoshLind — APPROVED** [review]

File: crates/aptos-infallible/src/mutex.rs [hunk 2]
```diff
@@ -30,6 +30,12 @@ impl<T> Mutex<T> {
     }
 }
 
+impl<T> Default for Mutex<Option<T>> {
```
Comment (sitalkedia) [inline crates/aptos-infallible/src/mutex.rs:33]: Unrelated?

**Review by sitalkedia — APPROVED** [review]

File: crates/aptos-rest-client/src/lib.rs (hunk 1 above)
Comment (gregnazario) [reply crates/aptos-rest-client/src/lib.rs:624]: This and a few other messages will need to be standardized and cleaned up, but this is fine.

**Review by gregnazario — APPROVED** [review]
